<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1111</width>
    <height>780</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Шифрование</string>
       </attribute>
       <widget class="QPushButton" name="Generate">
        <property name="geometry">
         <rect>
          <x>650</x>
          <y>420</y>
          <width>151</width>
          <height>41</height>
         </rect>
        </property>
        <property name="text">
         <string>Генерировать</string>
        </property>
       </widget>
       <widget class="QTableWidget" name="pbox">
        <property name="geometry">
         <rect>
          <x>750</x>
          <y>40</y>
          <width>321</width>
          <height>351</height>
         </rect>
        </property>
        <property name="lineWidth">
         <number>0</number>
        </property>
        <property name="showGrid">
         <bool>true</bool>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
        <attribute name="horizontalHeaderVisible">
         <bool>false</bool>
        </attribute>
        <attribute name="horizontalHeaderDefaultSectionSize">
         <number>159</number>
        </attribute>
        <attribute name="horizontalHeaderStretchLastSection">
         <bool>true</bool>
        </attribute>
        <attribute name="verticalHeaderVisible">
         <bool>false</bool>
        </attribute>
        <column>
         <property name="text">
          <string>Новый столбец</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Новый столбец</string>
         </property>
        </column>
       </widget>
       <widget class="QLabel" name="label_6">
        <property name="geometry">
         <rect>
          <x>760</x>
          <y>20</y>
          <width>141</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>P-блок</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_7">
        <property name="geometry">
         <rect>
          <x>380</x>
          <y>20</y>
          <width>141</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>S-блок</string>
        </property>
       </widget>
       <widget class="QTableWidget" name="sbox">
        <property name="geometry">
         <rect>
          <x>375</x>
          <y>40</y>
          <width>331</width>
          <height>351</height>
         </rect>
        </property>
        <property name="autoScroll">
         <bool>true</bool>
        </property>
        <attribute name="horizontalHeaderVisible">
         <bool>false</bool>
        </attribute>
        <attribute name="horizontalHeaderDefaultSectionSize">
         <number>164</number>
        </attribute>
        <attribute name="horizontalHeaderHighlightSections">
         <bool>true</bool>
        </attribute>
        <attribute name="horizontalHeaderStretchLastSection">
         <bool>true</bool>
        </attribute>
        <attribute name="verticalHeaderVisible">
         <bool>false</bool>
        </attribute>
        <column>
         <property name="text">
          <string>Новый столбец</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Новый столбец</string>
         </property>
        </column>
       </widget>
       <widget class="QGroupBox" name="groupBox">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>30</y>
          <width>271</width>
          <height>271</height>
         </rect>
        </property>
        <property name="title">
         <string>Данные</string>
        </property>
        <widget class="QLineEdit" name="j">
         <property name="geometry">
          <rect>
           <x>50</x>
           <y>190</y>
           <width>113</width>
           <height>20</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="m">
         <property name="geometry">
          <rect>
           <x>50</x>
           <y>140</y>
           <width>113</width>
           <height>20</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="label_3">
         <property name="geometry">
          <rect>
           <x>50</x>
           <y>120</y>
           <width>101</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Размер S блока:</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_4">
         <property name="geometry">
          <rect>
           <x>50</x>
           <y>170</y>
           <width>121</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Количество раундов:</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="n">
         <property name="geometry">
          <rect>
           <x>50</x>
           <y>90</y>
           <width>113</width>
           <height>20</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="label_2">
         <property name="geometry">
          <rect>
           <x>50</x>
           <y>70</y>
           <width>121</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Количество S блоков:</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_9">
         <property name="geometry">
          <rect>
           <x>50</x>
           <y>20</y>
           <width>47</width>
           <height>13</height>
          </rect>
         </property>
         <property name="text">
          <string>Логин:</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="login">
         <property name="geometry">
          <rect>
           <x>50</x>
           <y>40</y>
           <width>113</width>
           <height>20</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="err_dan">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>230</y>
           <width>211</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </widget>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>310</y>
          <width>271</width>
          <height>271</height>
         </rect>
        </property>
        <property name="title">
         <string>Ввод/вывод</string>
        </property>
        <widget class="QLabel" name="label">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>80</y>
           <width>131</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Импорт значений (путь):</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_5">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>150</y>
           <width>141</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Экспорт значений (путь):</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="file_export">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>170</y>
           <width>151</width>
           <height>20</height>
          </rect>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QLabel" name="error1">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>40</y>
           <width>281</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
        <widget class="QCheckBox" name="checkBox">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>200</y>
           <width>161</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Записать разность и пары</string>
         </property>
        </widget>
        <widget class="QPushButton" name="import_OK">
         <property name="geometry">
          <rect>
           <x>170</x>
           <y>100</y>
           <width>81</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Открыть...</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="file_import">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>100</y>
           <width>151</width>
           <height>20</height>
          </rect>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QPushButton" name="export_OK">
         <property name="geometry">
          <rect>
           <x>170</x>
           <y>170</y>
           <width>81</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Сохранить...</string>
         </property>
        </widget>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Анализ</string>
       </attribute>
       <widget class="QTableWidget" name="tableWidget_4">
        <property name="geometry">
         <rect>
          <x>560</x>
          <y>50</y>
          <width>491</width>
          <height>321</height>
         </rect>
        </property>
        <attribute name="horizontalHeaderDefaultSectionSize">
         <number>85</number>
        </attribute>
        <attribute name="horizontalHeaderStretchLastSection">
         <bool>true</bool>
        </attribute>
       </widget>
       <widget class="QPushButton" name="pushButton_2">
        <property name="geometry">
         <rect>
          <x>940</x>
          <y>370</y>
          <width>111</width>
          <height>41</height>
         </rect>
        </property>
        <property name="text">
         <string>Анализировать</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_8">
        <property name="geometry">
         <rect>
          <x>560</x>
          <y>10</y>
          <width>231</width>
          <height>16</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Таблица анализа блока замены</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_10">
        <property name="geometry">
         <rect>
          <x>530</x>
          <y>200</y>
          <width>21</width>
          <height>16</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string>ΔA</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_11">
        <property name="geometry">
         <rect>
          <x>800</x>
          <y>30</y>
          <width>21</width>
          <height>16</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string>ΔC</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_12">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>20</y>
          <width>31</width>
          <height>16</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string>ΔA:</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="deltaA">
        <property name="geometry">
         <rect>
          <x>50</x>
          <y>20</y>
          <width>191</width>
          <height>20</height>
         </rect>
        </property>
       </widget>
       <widget class="QTableWidget" name="xy">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>130</y>
          <width>441</width>
          <height>231</height>
         </rect>
        </property>
        <attribute name="horizontalHeaderDefaultSectionSize">
         <number>146</number>
        </attribute>
        <attribute name="horizontalHeaderStretchLastSection">
         <bool>true</bool>
        </attribute>
        <attribute name="verticalHeaderVisible">
         <bool>false</bool>
        </attribute>
        <attribute name="verticalHeaderStretchLastSection">
         <bool>false</bool>
        </attribute>
        <row>
         <property name="text">
          <string>Новая строка</string>
         </property>
        </row>
        <column>
         <property name="text">
          <string>№</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>X</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Y</string>
         </property>
        </column>
       </widget>
       <widget class="QTableWidget" name="xyc">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>360</y>
          <width>441</width>
          <height>231</height>
         </rect>
        </property>
        <attribute name="horizontalHeaderDefaultSectionSize">
         <number>110</number>
        </attribute>
        <attribute name="horizontalHeaderStretchLastSection">
         <bool>true</bool>
        </attribute>
        <attribute name="verticalHeaderVisible">
         <bool>false</bool>
        </attribute>
        <row>
         <property name="text">
          <string>Новая строка</string>
         </property>
        </row>
        <column>
         <property name="text">
          <string>№</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>X'</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Y'</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>ΔC</string>
         </property>
        </column>
       </widget>
       <widget class="QLabel" name="label_13">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>50</y>
          <width>121</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Количество текстов:</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="kol_text">
        <property name="geometry">
         <rect>
          <x>130</x>
          <y>50</y>
          <width>113</width>
          <height>20</height>
         </rect>
        </property>
       </widget>
       <widget class="QPushButton" name="Give_button">
        <property name="geometry">
         <rect>
          <x>250</x>
          <y>40</y>
          <width>81</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string>Получить</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_14">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>100</y>
          <width>81</width>
          <height>21</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Результат</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_15">
        <property name="geometry">
         <rect>
          <x>570</x>
          <y>508</y>
          <width>101</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Проверить ключ</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="key">
        <property name="geometry">
         <rect>
          <x>670</x>
          <y>508</y>
          <width>221</width>
          <height>20</height>
         </rect>
        </property>
       </widget>
       <widget class="QPushButton" name="key_button">
        <property name="geometry">
         <rect>
          <x>900</x>
          <y>500</y>
          <width>91</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string>Проверить</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_key">
        <property name="geometry">
         <rect>
          <x>680</x>
          <y>530</y>
          <width>231</width>
          <height>31</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
       <widget class="QProgressBar" name="progressBar">
        <property name="geometry">
         <rect>
          <x>690</x>
          <y>540</y>
          <width>201</width>
          <height>23</height>
         </rect>
        </property>
        <property name="value">
         <number>0</number>
        </property>
       </widget>
       <widget class="QLabel" name="len_key">
        <property name="geometry">
         <rect>
          <x>570</x>
          <y>470</y>
          <width>71</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Длина ключа:</string>
        </property>
       </widget>
       <widget class="QLabel" name="zna4_key">
        <property name="geometry">
         <rect>
          <x>650</x>
          <y>470</y>
          <width>21</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>1</string>
        </property>
       </widget>
       <widget class="QLabel" name="err3">
        <property name="geometry">
         <rect>
          <x>70</x>
          <y>80</y>
          <width>321</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
       <widget class="QLabel" name="err4">
        <property name="geometry">
         <rect>
          <x>700</x>
          <y>190</y>
          <width>241</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
       <widget class="QLabel" name="label_16">
        <property name="geometry">
         <rect>
          <x>590</x>
          <y>430</y>
          <width>271</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1111</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
